### DISCUSSION TITLE
Grid grouped on load

### DISCUSSION THREAD

**Discussion by osdan-madrid** [serenity-is/Serenity#6408] [Help] [answered]
Hi everyone,

I've 2 quesstions, maybe someone can help me

I've grouped by Client,

![image](https://user-images.githubusercontent.com/42551335/171504434-20d6cdc6-3b24-486b-96d0-347f099dc442.png)

The first question, is it possible to load the page with the info with the info collapsed? Something like this:

![image](https://user-images.githubusercontent.com/42551335/171504883-75e3a3e1-bd5a-4dcf-8e57-000528001b7c.png)

The second question, is it possible to add the sum value next to the count values in grouped header?

![image](https://user-images.githubusercontent.com/42551335/171505146-3ddcd7c4-1c14-4672-8093-c69cf13b8bac.png)

Thanks in advance !! :D

**Comment by minhhungit** [comment]
first question, yes you can, try to use this code:

![image](https://user-images.githubusercontent.com/2279508/171511893-09124c0e-00cc-497c-93ea-4eefdd55c2ea.png)

second question, you can too, do the same what you counted item (on your image your counted '3 items' ) - use this.view.getItems and filter them by groupping/parent row

**Reply by osdan-madrid** [reply]
Thanks @minhhungit  !!

The first question works perfect!

But I've problems with your suggestion for the second question, I don't know how to sum the filtered data :(


`            this.view.setGrouping(
                [{
                    
                    //formatter: x => '<b>CLIENTE: ' + x.value + ' (' + x.count ' item(s) )</b>',
                    formatter: x => '<b>CLIENTE: ' + x.value + ' (' + x.count + ' item(s)' + this.view.getItems().filter(y => y.ClienteNombre == x.value) + ' )</b>',
                    getter: 'ClienteNombre',
                    aggregators: [
                        new Slick.Aggregators.Sum('Precio'),
                    ]
                    
                }])`

Thanks in advance

**Comment by minhhungit** [comment]
and be careful to sum value using `this.view` if you use paging, because you might miss records

**Reply by osdan-madrid** [reply]
Thanks !!

I use to avoid Pager




`        protected usePager() {
            return false;
        }
`

**Comment by minhhungit** [comment, marked as answer]
Change your formatter from 
```javascript
formatter: x => '<b>CLIENTE: ' + x.value + ' (' + x.count + ' item(s)' + this.view.getItems().filter(y => y.ClienteNombre == x.value) + ' )</b>'
```

to

```javascript
formatter: x => {
    let result = "<b>CLIENTE: ' + x.value + ' (' + x.count + ' item(s)'</b>";
    var sum = this.view.getItems().filter(y => y.ClienteNombre == x.value).reduce((m, n) => m + n["YOUR_FIELD"], 0);
    result  = result + 'Sum: ' + sum;

    return result;
}
```

**Reply by osdan-madrid** [reply]
It works perfectly !!

Thanks!!
